webpay/auth/views.py

```diff
@@ -177,28 +212,31 @@ def _fxa_authorize(fxa, client_secret, request, auth_response):
         urlparse.urljoin(settings.FXA_OAUTH_URL, 'v1/verify'),
         data=json.dumps({'token': token['access_token']}),
         headers={'Content-Type': 'application/json'})
-    return res.json()
+    return res.json(), token
 
 
 @anonymous_csrf_exempt
 @json_view
 def fxa_login(request):
     session = get_fxa_session(state=request.POST.get('state'))
-    data = _fxa_authorize(
+    result, token = _fxa_authorize(
         session,
         settings.FXA_CLIENT_SECRET,
         request,
         request.POST.get('auth_response'))
-    log.info("FxA login response: " + repr(data))
 
-    if data.get('error'):
+    log.info("FxA token verification response. result={r}; token={t}"
+             .format(r=result, t=token))
+
+    if result.get('error'):
         # All errors look like:
         # https://github.com/mozilla/fxa-oauth-server/blob/master
         # /docs/api.md#errors
-        log.info('FxA login error: {data}'.format(data=data))
+        log.info('FxA login error: {r}'.format(r=result))
         return http.HttpResponseForbidden()
 
-    user_hash = set_user(request, data['email'], verified=True)
+    _process_fxa_auth_ts(token, request)
+    user_hash = set_user(request, result['email'], verified=False)
```

Comment:
I think the `verified=False` bit needs to be revisited. It should likely be removed from the call and the `set_user` function.